test/hotspot/jtreg/compiler/vectorapi/VectorMaskCompareNotTest.java

```diff
@@ -0,0 +1,693 @@
+        ByteVector av = ByteVector.fromArray(B_SPECIES, ba, 0);
+        ByteVector bv = ByteVector.fromArray(B_SPECIES, bb, 0);
+        VectorMask<Byte> m1 = av.compare(op, bv);
+        func.apply(m1).intoArray(mr, 0);
+
+        for (int i = 0; i < B_SPECIES.length(); i++) {
+            verifyResults(ba[i], bb[i], mr[i], op);
+        }
+    }
+
+    @ForceInline
+    public static void testCompareMaskNotShort(VectorOperators.Comparison op, VectorMaskOperator func) {
+        ShortVector av = ShortVector.fromArray(S_SPECIES, sa, 0);
+        ShortVector bv = ShortVector.fromArray(S_SPECIES, sb, 0);
+        VectorMask<Short> m1 = av.compare(op, bv);
+        func.apply(m1).intoArray(mr, 0);
+
+        for (int i = 0; i < S_SPECIES.length(); i++) {
+            verifyResults(sa[i], sb[i], mr[i], op);
+        }
+    }
+
+    @ForceInline
+    public static void testCompareMaskNotInt(VectorOperators.Comparison op, VectorMaskOperator func) {
+        IntVector av = IntVector.fromArray(I_SPECIES, ia, 0);
+        IntVector bv = IntVector.fromArray(I_SPECIES, ib, 0);
+        VectorMask<Integer> m1 = av.compare(op, bv);
+        func.apply(m1).intoArray(mr, 0);
+
+        for (int i = 0; i < I_SPECIES.length(); i++) {
+            verifyResults(ia[i], ib[i], mr[i], op);
+        }
+    }
+
+    @ForceInline
+    public static void testCompareMaskNotLong(VectorOperators.Comparison op, VectorMaskOperator func) {
+        LongVector av = LongVector.fromArray(L_SPECIES, la, 0);
+        LongVector bv = LongVector.fromArray(L_SPECIES, lb, 0);
+        VectorMask<Long> m1 = av.compare(op, bv);
+        func.apply(m1).intoArray(mr, 0);
+
+        for (int i = 0; i < L_SPECIES.length(); i++) {
+            verifyResults(la[i], lb[i], mr[i], op);
+        }
+    }
+
+    @ForceInline
+    public static void testCompareMaskNotFloat(VectorOperators.Comparison op, float[] a, float[] b, VectorMaskOperator func) {
+        FloatVector av = FloatVector.fromArray(F_SPECIES, a, 0);
+        FloatVector bv = FloatVector.fromArray(F_SPECIES, b, 0);
+        VectorMask<Float> m1 = av.compare(op, bv);
+        func.apply(m1).intoArray(mr, 0);
+
+        for (int i = 0; i < F_SPECIES.length(); i++) {
+            verifyResults(a[i], b[i], mr[i], op);
+        }
+    }
+
+    @ForceInline
+    public static void testCompareMaskNotDouble(VectorOperators.Comparison op, double[] a, double[] b, VectorMaskOperator func) {
+        DoubleVector av = DoubleVector.fromArray(D_SPECIES, a, 0);
+        DoubleVector bv = DoubleVector.fromArray(D_SPECIES, b, 0);
+        VectorMask<Double> m1 = av.compare(op, bv);
+        func.apply(m1).intoArray(mr, 0);
+
+        for (int i = 0; i < D_SPECIES.length(); i++) {
+            verifyResults(a[i], b[i], mr[i], op);
+        }
+    }
+
+    // Byte tests
+    @Test
+    @IR(counts = { IRNode.XOR_V_MASK, "= 0", IRNode.XOR_VB, "= 0" },
```

Comment:
Could you still assert the presence of some other vectors, just to make sure we are indeed getting vectors here?